File: myCTCA Mobile/Android/src/myctca/app/src/main/java/com/myctca/model/ExternalLink.java
package com.myctca.model;

import java.util.List;

public class ExternalLink {
    private String title;
    private String url;
    private boolean isDownload;
    private List<ExternalLink> children;

    public String getTitle() {
        return title;
    }

    public boolean getIsDownload() {
        return isDownload;
    }

    public String getUrl() {
        return url;
    }

    public List<ExternalLink> getChildren() {
        return children;
    }
}

File: myCTCA Mobile/iOS/src/myctca/Models/ExternalLink.swift
//
//  ExternalLink.swift
//  myctca
//
//  Created by Manjunath K on 4/14/21.
//  Copyright © 2021 CTCA. All rights reserved.
//

import Foundation

class ExternalLinkItem: Codable {
    var title: String = ""
    var url: String = ""
    var isDownload: Bool = false

    enum CodingKeys: String, CodingKey {
        case title, url, isDownload
    }
}

class ExternalLink: Codable {
    var title: String = ""
    var url: String = ""
    var isDownload: Bool = false
    var items:[ExternalLinkItem]?
    
    enum CodingKeys: String, CodingKey {
        case title, url, isDownload
        case items = "children"
    }
}
